From 97d01b91031fd3d8a9a31b4657dd5596c0311b55 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=90=D0=BB=D0=B5=D0=BA=D1=81=D0=B5=D0=B9?=
 <alexey.pawlow@gmail.com>
Date: Thu, 17 Jun 2021 18:52:09 +0530
Subject: [PATCH 100/N] build: Fix ncursesw include lookup

Mirror what is already done for libffi; Look it up via
pkg-config and use the exported path via sysconfig in setup.py

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 Makefile.pre.in | 1 +
 configure.ac    | 9 ++++++++-
 setup.py        | 2 +-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index e526be2..da1d638 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -319,6 +319,7 @@ IO_OBJS=	\
 ##########################################################################
 
 LIBFFI_INCLUDEDIR=	@LIBFFI_INCLUDEDIR@
+NCURSESW_INCLUDEDIR=	@NCURSESW_INCLUDEDIR@
 
 ##########################################################################
 # Parser
diff --git a/configure.ac b/configure.ac
index ca7e7a2..ee6db8e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5603,10 +5603,17 @@ then
   [Define if you have struct stat.st_mtimensec])
 fi
 
+if test -n "$PKG_CONFIG"; then
+    NCURSESW_INCLUDEDIR="`"$PKG_CONFIG" ncursesw --cflags-only-I 2>/dev/null | sed -e 's/^-I//;s/ *$//'`"
+else
+    NCURSESW_INCLUDEDIR=""
+fi
+AC_SUBST(NCURSESW_INCLUDEDIR)
+
 # first curses header check
 ac_save_cppflags="$CPPFLAGS"
 if test "$cross_compiling" = no; then
-  CPPFLAGS="$CPPFLAGS -I/usr/include/ncursesw"
+  CPPFLAGS="$CPPFLAGS -I$NCURSESW_INCLUDEDIR"
 fi
 
 AC_CHECK_HEADERS(curses.h ncurses.h)
diff --git a/setup.py b/setup.py
index 5eb9b07..67e290a 100644
--- a/setup.py
+++ b/setup.py
@@ -1192,7 +1192,7 @@ class PyBuildExt(build_ext):
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
             if not CROSS_COMPILING:
-                curses_includes.append('/usr/include/ncursesw')
+                curses_includes.append(sysconfig.get_config_var("NCURSESW_INCLUDEDIR"))
             # Bug 1464056: If _curses.so links with ncursesw,
             # _curses_panel.so must link with panelw.
             panel_library = 'panelw'
-- 
2.34.1

